tests/test_multiple_files.py

```diff
@@ -0,0 +1,46 @@
+import tempfile
+from unittest import mock
+
+import xarray as xr
+
+from metofficedatahub.multiple_files import MetOfficeDataHub, save_to_zarr
+from tests.conftest import mocked_requests_get
+
+
+def test_init():
+    _ = MetOfficeDataHub(client_id="fake", client_secret="fake")
+
+
+@mock.patch("requests.get", side_effect=mocked_requests_get)
+def test_download_all_none(mock_get):
+    """Check that if there are no order ids, then no data is downloaded"""
+    datahub = MetOfficeDataHub(client_id="fake", client_secret="fake")
```

Comment:
is there a `beforeEach` or something like that in python that we can call instead?

I'm not sure, but I think you could probably do something like:

```python
class DataHubTestCase(unittest.TestCase):
    def setUp(self):
        self.datahub = MetOfficeDataHub(client_id="fake", client_secret="fake")

    def test_download_all_none(mock_get):
        self.datahub.download_all_files(...)
```

see [`setUp`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.setUp) and [unittest > Organizing tests](https://docs.python.org/3/library/unittest.html#organizing-tests)

Reply:
Yea good call, can infact do this with pytest fixtures